x/http_secure_redirect.go

```diff
@@ -55,6 +55,14 @@ func SecureRedirectOverrideDefaultReturnTo(defaultReturnTo *url.URL) SecureRedir
 	}
 }
 
+// SecureRedirectToIsWhitelisted validates if the redirect_to param is allowed for a given wildcard
+func SecureRedirectToIsWhiteListedHost(returnTo *url.URL, allowed url.URL) bool {
+	if allowed.Host != "" && allowed.Host[:1] == "*" {
+		return strings.HasSuffix(returnTo.Host, allowed.Host[1:])
```

Comment:
What's the plan for case sensitivity here?
- On documentation, the function docstring is ambiguous,
- when wildcards are used, the match is case-sensitive,
- when wildcards are not used, the match is case-insensitive.

To follow the principle of least surprise, you might want to do one of
- Do strings.HasSuffix(strings.ToLower(returnTo.Host), strings.ToLower(allowed.Host[1:]) here to be case-insensitive.
- .. or do return allowed.Host == returnTo.Host few lines later to be case-sensitive.
- .. or mention this semi-case-insensitivity behavior in the function comment if it is intentional and preferred.

Reply:
Hostnames are case insensitive, so the correct approach would be to have the wildcard matches also case insensitive :)